target/linux/x86/base-files/usr/sbin/grow_rootfspart

```diff
@@ -0,0 +1,327 @@
+	#mbr_first_chs=${partentry_hex:MBR_PARTFIRST_CHS*2:MBR_PARTCHS_SIZE*2}
+	#mbr_type=${partentry_hex:MBR_PARTTYPE*2:MBR_PARTTYPE_SIZE*2}
+	#mbr_last_chs=${partentry_hex:MBR_PARTLAST_CHS*2:MBR_PARTCHS_SIZE*2}
+	mbr_mbr_first_sector_hex_le=${partentry_hex:MBR_PARTFIRST_SECTOR*2:MBR_PARTSECTOR_SIZE*2}
+	mbr_first_sector_hex=$(bswap "$mbr_mbr_first_sector_hex_le")
+	mbr_mbr_number_of_sectors_hex_le=${partentry_hex:MBR_PARTNUMBER_OF_SECTOR*2:MBR_PARTSECTOR_SIZE*2}
+	mbr_number_of_sectors_hex=$(bswap "$mbr_mbr_number_of_sectors_hex_le")
+}
+
+gpt_parse_part(){
+	local partentry_hex=$1
+
+        gpt_partition_type=${partentry_hex:GPT_PARTITION_TYPE*2:GPT_PARTITION_TYPE_SIZE*2}
+        [ "$gpt_partition_type" != "00000000000000000000000000000000" ] || return 1
+
+        gpt_partition_first_lba_hex_le=${partentry_hex:GPT_PARTITION_FIRST_LBA*2:GPT_LBA_SIZE*2}
+        gpt_partition_first_lba_hex=$(bswap "$gpt_partition_first_lba_hex_le")
+        gpt_partition_last_lba_hex_le=${partentry_hex:GPT_PARTITION_LAST_LBA*2:GPT_LBA_SIZE*2}
+        gpt_partition_last_lba_hex=$(bswap "$gpt_partition_last_lba_hex_le")
+        #gpt_partition_attribute_hex_le=${partentry_hex:GPT_PARTITION_ATTRIBUTE*2:GPT_PARTITION_ATTRIBUTE_SIZE*2}
+        #gpt_partition_label_utf16le=${partentry_hex:GPT_PARTITION_LABEL*2:GPT_PARTITION_LABEL_SIZE*2}
+}
+
+DISK_SECTOR=$((0x0200))
+
+#CHS conversion
+HPC=16
+SPT=63
+
+#MBR_BOOTSTRAP=0
+#MBR_BOOTSTRAP_SIZE=446
+MBR_PART1_ENTRY=$((0x01BE))
+MBR_PART2_ENTRY=$((0x01CE))
+MBR_PART3_ENTRY=$((0x01DE))
+MBR_PART4_ENTRY=$((0x01EE))
+MBR_PARTENTRY_SIZE=16
+
+#MBR_PARTSTATUS=$((0x0)) MBR_PARTSTATUS_SIZE=1
+MBR_PARTFIRST_CHS=$((0x01)) MBR_PARTLAST_CHS=$((0x05)) MBR_PARTCHS_SIZE=3
+#MBR_PARTTYPE=$((0x04)) MBR_PARTTYPE_SIZE=1
+MBR_PARTFIRST_SECTOR=$((0x08)) MBR_PARTNUMBER_OF_SECTOR=$((0x0C)) MBR_PARTSECTOR_SIZE=4
+
+MBR_MAX_SECTORS=$((2**32-1))
+
+#GPT
+GPT_HEADER_LBA=1
+GPT_HEADER=$((GPT_HEADER_LBA*DISK_SECTOR))
+
+GPT_SIG=$((0x00)) GPT_SIG_SIZE=8
+GPT_HEADER_SIZE=$((0x0C)) GPT_HEADER_SIZE_SIZE=4
+GPT_HEADER_CRC32=$((0x10)) GPT_PARTITIONS_CRC32=$((0x58)) GPT_CRC32_SIZE=4
+GPT_CURRENT_LBA=$((0x18)) GPT_BACKUP_LBA=$((0x20)) GPT_LBA_SIZE=8
+GPT_FIRST_LBA=$((0x28)) GPT_LAST_LBA=$((0x30))
+GPT_PARTITIONS_LBA=$((0x48))
+GPT_NUMBER_OF_PARTITIONS=$((0x50)) GPT_NUMBER_OF_PARTITIONS_SIZE=4
+GPT_PARTITION_ENTRY_SIZE=$((0x54)) GPT_PARTITION_ENTRY_SIZE_SIZE=4
+
+GPT_PARTITION_TYPE=$((0x0)) GPT_PARTITION_TYPE_SIZE=16
+GPT_PARTITION_FIRST_LBA=$((0x20)) GPT_PARTITION_LAST_LBA=$((0x28))
+#GPT_PARTITION_ATTRIBUTE=$((0x38)) GPT_PARTITION_ATTRIBUTE_SIZE=8
+#GPT_PARTITION_LABEL=$((0x38)) GPT_PARTITION_LABEL_SIZE=72
+
+GPT_MAGIC="4546492050415254"
+
+
+disk=$1
+disk=$(readlink -f "$1")
+
+[ -r "$1" ] ||
+	die 1 "Failed to read '$1'"
+
+if [ -b "$disk" ]; then
+        disk_major_minor=$(ls -l "$disk" | awk '{ print substr($5,1,length($5)-1) ":" $6 }')
+        disk_size_sectors=$(($(cat "/sys/dev/block/$disk_major_minor/size")))
```

Comment:
From this point, indent start to be broken regularly again.

Reply:
It is fixed now.